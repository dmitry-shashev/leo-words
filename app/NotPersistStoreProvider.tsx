'use client'

import { FC, ReactNode, useEffect } from 'react'
import { Provider } from 'react-redux'
import { createStore } from '@/store/store'
import packageJson from '../package.json'

interface Props {
  children: ReactNode
}

export const NotPersistStoreProvider: FC<Props> = ({ children }) => {
  useEffect(() => {
    const currenVersion = sessionStorage.getItem('version')
    const currentUrl = sessionStorage.getItem('currentUrl')

    if (
      !currenVersion ||
      currenVersion !== packageJson.version ||
      currentUrl !== location.href
    ) {
      sessionStorage.setItem('currentUrl', location.href)
      sessionStorage.setItem('version', packageJson.version)
      window.location.reload()
      return
    }
  }, [])

  return <Provider store={createStore()}>{children}</Provider>
}
