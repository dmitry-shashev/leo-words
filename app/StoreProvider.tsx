'use client'

import { FC, ReactNode } from 'react'
import { Provider } from 'react-redux'
import { persistor, store } from '@/store/store'
import { PersistGate } from 'redux-persist/integration/react'

interface Props {
  children: ReactNode
}

export const StoreProvider: FC<Props> = ({ children }) => {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistor}>{children}</PersistGate>
    </Provider>
  )
}
