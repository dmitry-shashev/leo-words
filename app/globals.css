@tailwind base;
@tailwind components;
@tailwind utilities;

@keyframes delayed-appearing-animation {
  from {
    opacity: 0;
  }
  to {
    opacity: 0;
  }
}

.delayedAppearing {
  animation: delayed-appearing-animation 0.4s;
}
