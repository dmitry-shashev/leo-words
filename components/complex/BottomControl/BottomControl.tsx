'use client'

import { FC } from 'react'
import { RegularBtn } from '@/components/simple/RegularBtn/RegularBtn'
import { TransparentBtn } from '@/components/simple/TransparentBtn/TransparentBtn'
import { useSelector } from 'react-redux'
import {
  checkNo,
  checkOk,
  getIsByWrongWords,
  getIsCheckState,
  goBack,
  goToCheckState,
} from '@/store/slices/wordsSlice'
import { useAppDispatch } from '@/store/store'

export const BottomControl: FC = () => {
  const dispatch = useAppDispatch()
  const isCheckState = useSelector(getIsCheckState)
  const isByWrongWords = useSelector(getIsByWrongWords)

  const onCheck = (): void => {
    dispatch(goToCheckState())
  }

  const onOk = (): void => {
    dispatch(checkOk())
  }

  const onNo = (): void => {
    dispatch(checkNo())
  }

  const onBack = (): void => {
    dispatch(goBack())
  }

  return (
    <div>
      <div className="flex justify-between gap-4 mb-8">
        {isCheckState ? (
          <>
            <RegularBtn onClick={onNo} label="No" color="red" />
            <RegularBtn onClick={onOk} label="Ok" color="green" />
          </>
        ) : (
          <RegularBtn onClick={onCheck} label="Check" color="blue" />
        )}
      </div>
      <div className="flex">
        {!isByWrongWords && <TransparentBtn onClick={onBack} label="Back" />}
      </div>
    </div>
  )
}
