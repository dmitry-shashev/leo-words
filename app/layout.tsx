import type { Metadata } from 'next'
import { Inter } from 'next/font/google'
import './globals.css'
import { ReactNode } from 'react'
import { StoreProvider } from '@/app/StoreProvider'

const inter = Inter({ subsets: ['latin'] })

export const metadata: Metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}

export default function RootLayout({
  children,
}: Readonly<{
  children: ReactNode
}>): ReactNode {
  return (
    <html lang="en">
      <body className={inter.className}>
        <div className="p-2">
          <StoreProvider>{children}</StoreProvider>
        </div>
      </body>
    </html>
  )
}
