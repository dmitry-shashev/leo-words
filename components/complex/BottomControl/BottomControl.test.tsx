import { expect, test } from 'vitest'
import { render, fireEvent } from '@testing-library/react'
import { BottomControl } from '@/components/complex/BottomControl/BottomControl'
import { StoreProvider } from '@/app/StoreProvider'

test('BottomControl', async () => {
  const { asFragment, getByLabelText, queryByLabelText } = render(
    <StoreProvider>
      <BottomControl />
    </StoreProvider>
  )

  function checkButtons(
    ok: boolean,
    no: boolean,
    check: boolean,
    back: boolean
  ): void {
    if (ok) {
      expect(queryByLabelText('Ok')).not.toBeNull()
    } else {
      expect(queryByLabelText('Ok')).toBeNull()
    }
    if (no) {
      expect(queryByLabelText('No')).not.toBeNull()
    } else {
      expect(queryByLabelText('No')).toBeNull()
    }
    if (check) {
      expect(queryByLabelText('Check')).not.toBeNull()
    } else {
      expect(queryByLabelText('Check')).toBeNull()
    }
    if (back) {
      expect(queryByLabelText('Back')).not.toBeNull()
    } else {
      expect(queryByLabelText('Back')).toBeNull()
    }
  }

  checkButtons(false, false, true, true)
  fireEvent.click(getByLabelText('Check'))

  checkButtons(true, true, false, true)
  fireEvent.click(getByLabelText('Ok'))

  checkButtons(false, false, true, true)
  fireEvent.click(getByLabelText('Check'))

  checkButtons(true, true, false, true)
  fireEvent.click(getByLabelText('No'))

  checkButtons(false, false, true, true)
  fireEvent.click(getByLabelText('Back'))

  expect(asFragment()).toMatchSnapshot()
})
