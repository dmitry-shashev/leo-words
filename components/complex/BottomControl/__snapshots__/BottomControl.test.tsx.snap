// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BottomControl 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-between gap-4 mb-8"
    >
      <button
        aria-label="Check"
        class="bg-blue-500 text-3xl h-16  font-semibold text-white rounded grow"
        type="button"
      >
        Check
      </button>
    </div>
    <div
      class="flex"
    >
      <button
        aria-label="Back"
        class="grow text-lg text-blue-200 border-blue-200 border rounded h-10"
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</DocumentFragment>
`;
